blockchain/contracts/WavGame.sol

```diff
@@ -0,0 +1,270 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import "@openzeppelin/contracts/utils/introspection/ERC165.sol";
+import "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import "@opengsn/contracts/src/ERC2771Recipient.sol";
+import "@opengsn/contracts/src/interfaces/IERC2771Recipient.sol";
+import "./interfaces/IWavGame.sol";
+import "./interfaces/IWavContract.sol";
+import "./libraries/Helper.sol";
+import "hardhat/console.sol";
+
+
+contract WavGame is IWavGame, Ownable, ERC2771Recipient, ReentrancyGuard, ERC165 {
+    using EnumerableSet for EnumerableSet.UintSet;
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    uint256 public entryFee; // In wei
+    uint256 constant internal ENTRY_LEVEL = 1; 
+    IWavContract public wavContract; 
+    address payable public revenueContract; // Platform revenue spliting contract
+
+    mapping(address => IWavGame.Level[]) internal wavGames; // Levels are 0-indexed. i.e level 1 = 0 in level
+    mapping(address => mapping(uint256 => EnumerableSet.AddressSet)) collectors;// Collectors per level per owner game
+    mapping(address => mapping(uint256 => EnumerableSet.UintSet)) burnableSet;// Burnable IDs per level per owner game
+    mapping(address => mapping(uint256 => EnumerableSet.UintSet)) mintableSet;// Mintable IDs per level per owner game
+
+    
+    event Collected(address caller, address to, uint256 amount, uint256 refunded, uint256 totalMinted);
+    event LeveledUp(address caller, address collector, uint256 nextLevel, uint256 totalMinted);
+    event SpecialMint(address caller, address collector, uint256 id, uint256 amount);
+    event PaymentForwarded(address to, uint256 amount);
+    event GameSet(address owner, uint256 level);
+    event LevelSet(address owner, uint256 level);
+
+	// Create all initial artists
+	// create and map the 6 artist level / island to each artist, and define the Levels specific details per artist
+    constructor(IWavContract _wavContract, address _trustedForwarder,uint256 _entryFee, address payable _revenueContract) {
```

Comment:
Can we rename `WavContract` to something like `WavNFT` across the contracts? This will make it easier for us to identify tokens apart from game logic.